tests/posix/pmemfile_test.hpp

```diff
@@ -151,6 +154,16 @@ class pmemfile_test : public testing::Test {
 	      poolsize(poolsize),
 	      test_empty_dir_on_teardown(true)
 	{
+#ifdef FAULT_INJECTION
+		fault_injection = true;
+#else
+		fault_injection = false;
+#endif
+		char *is_pmemfile_pop_str = std::getenv("LIBPMEMFILE_POP");
```

Comment:
I think there should be no need for another environment variable for this.
Being linked with libpmemfile-pop is a decision already made when linking, it doesn't depend on the environment at runtime.

How about this:
Just add another symbol next to `pmemfile_inject_fault_at`, like `bool is_fault_injection_enabled`, which tells you if it is enabled at runtime. If it is disabled at compile time already, then just add another dummy library to the list of libraries linked with all tests, which does this:
```
bool is_fault_injection_enabled(void)
{
   return false; /* disabled at build time already */
}

pmemfile_inject_fault_at(..)
{
    abort();
}
```
Alternatively, these two dummy functions can be defined  in `pmemfile_test.hpp`:
```
#ifndef FAULT_INJECTION
/* symbols are not exposed by the tested library, mock them here */
static inline bool is_fault_injection_enabled(void)
{
   return false;
}

#endif
```